### PR TITLE
Implement \R, \v, \h for character/scalar modes

### PR DESCRIPTION
**Pull request by natecook1000**
This implements these meta characters as per the Unicode proposal. We still need to expand how we treat `\n`. It should have mostly the same behavior as `\v`, but I also want to make sure that `/\r\n/` will match an `"\r\n"` character.

### CODE REVIEW

**Comment by natecook1000** [comment]
@swift-ci Please test

**Review by hamishknight — APPROVED** [review]
Thanks! Do you also want to take https://github.com/apple/swift-experimental-string-processing/issues/375 to fix the handling of `\N`?

**Review by Azoy — APPROVED** [review]

File: Sources/_StringProcessing/Unicode/ScalarProps.swift [hunk 1]
```diff
@@ -46,3 +46,19 @@ extension Unicode.Script {
...
+  var isVerticalWhitespace: Bool {
+    switch value {
+      case 0x000A...0x000D /* LF ... CR */: return true
+      case 0x0085 /* NEXT LINE (NEL) */: return true
+      case 0x2028 /* LINE SEPARATOR */: return true
+      case 0x2029 /* PARAGRAPH SEPARATOR */: return true
+      default: return false
+    }
+  }
+}
```
Comment (milseman) [inline Sources/_StringProcessing/Unicode/ScalarProps.swift:64]: Character has a `isNewline`. Can we share implementation and/or SPI for this?

Comment (natecook1000) [reply Sources/_StringProcessing/Unicode/ScalarProps.swift:64]: It's the same list — we should ultimately have this on UnicodeScalar as well; probably with a name to match `Character`'s.

**Comment by natecook1000** [comment]
@swift-ci Please test

**Comment by natecook1000** [comment]
@swift-ci Please test

**Comment by natecook1000** [comment]
@swift-ci Please test

**Comment by natecook1000** [comment]
@swift-ci Please test

**Comment by natecook1000** [comment]
@swift-ci Please test macOS platform

**Comment by natecook1000** [comment]
@swift-ci Please test macOS platform

**Comment by natecook1000** [comment]
Figured this issue out — we were looking at the character view to test whether the current position was the start of a line, even in Unicode scalar mode. I've fixed that and audited the rest of the assertions to make sure that we're looking at the correct view.

**Comment by natecook1000** [comment]
@swift-ci Please test

**Comment by natecook1000** [comment]
@swift-ci Please test

**Pull request merged**